### ISSUE TITLE
Select Component - unable to interact with or select options using VoiceOver

### ISSUE THREAD

**Issue by pierscm** [uber/baseweb#5282]
When using VoiceOver, the select component can not be interacted with or used to select an option when `searchable` is false. When `searchable` is true, VoiceOver does not provide sufficient information to the user.

Example can be seen in the BaseUI documentation: https://baseweb.design/components/select/#select

## Current Behavior

When `searchable` is false:
* Navigating by pressing tab, it is not apparent that the user is on a select component. VoiceOver reads “group” and provides incorrect controls that do not do anything: “to interact press control + option + shift + down arrow”.
* Navigating by pressing `control + option + space` , it is not apparent that the user is on a select component. VoiceOver reads "text element" and provides no additional information.

When `searchable` is true: 
* When navigating to the element VoiceOver reads the controls to interact `control + option + space`, but once the menu is open there is no additional information provided to the user about how to navigate through or select options. When pressing up or down arrows, every option is incorrectly read as "selected"

## Expected Behavior

For both searchable states, a user using VoiceOver should be provided information about how to interact with the element and select an option.

[W3Example](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/)

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v12.2.0  |
| React   |   18.2.0      |
| browser |    Chrome     |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.
